modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxPrepareFuture.java

```diff
@@ -1951,7 +1953,7 @@ void onResult(GridDhtTxPrepareResponse res) {
 
                 boolean rec = cctx.gridEvents().isRecordable(EVT_CACHE_REBALANCE_OBJECT_LOADED);
 
-                for (GridCacheEntryInfo info : res.preloadEntries()) {
+                for (GridCacheEntryInfo info : F.emptyIfNull(res.preloadEntries())) {
```

Comment:
If `preloadEntries` can be null, let put `@Nullable` around it, or the seggers, getters.